### PR TITLE
:sparkles: feat: add task 6

### PR DESCRIPTION
**Pull request by sigurdnf**

### CODE REVIEW

File: src/mongodb/database.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def __init__(
         user: str = "root",
         password: str = "password",
         port: int = 27017,
-        connection_opts: str = "?authSource=admin",
+        # connection_opts: str = "?authSource=admin",
```
Comment (larwaa) [inline src/mongodb/database.py]: Behold dette alternativet
```suggestion
        connection_opts: str = "?authSource=admin",
```

File: src/mongodb/tasks.py [hunk 2]
```diff
@@ -56,6 +60,44 @@ def task2(self):
...
+                {"$project": {"user_id": 1, "trackPointsCount": { "$size": "$track_points" }}},
+                
+                {"$group": {"_id": "$user_id", "averageTrackPointsPerUser": { "$avg": "$trackPointsCount" }}},
+            
+                {"$sort": {"_id": 1}}
+
+                
+            ])
+        
+        return pd.DataFrame(list(res))
```
Comment (larwaa) [inline src/mongodb/tasks.py]:
```suggestion
```

File: src/mongodb/tasks.py [hunk 3]
```diff
@@ -113,6 +155,101 @@ def task5(self):
...
+    def task6b(self):
+        """
+        6. 
+            b) Is this also the year with most recorded hours?
+        """
+        res = self.db.activities.aggregate([
+          
+        {"$project": {
+        "year": { "$year": "$start_datetime" },
+        "duration": { "$divide": [{ "$subtract": ["$end_datetime", "$start_datetime"] }, 3600000] }
```
Comment (larwaa) [inline src/mongodb/tasks.py:154]:
```suggestion
        "duration": { "$divide": [{ "$subtract": ["$end_datetime", "$start_datetime"] }, 60 * 60 * 1000] }
```

Comment (larwaa) [reply src/mongodb/tasks.py:154]: Nitpicking, can use `$dateDiff` instead?

File: src/mongodb/tasks.py [hunk 4]
```diff
@@ -113,6 +155,101 @@ def task5(self):
...
+#     },
+#     {
+#         "$sort": {"activityCount": -1}  # Sort by activityCount in descending order
+#     },
+#     {
+#         "$limit": 1  # Limit to the top result (year with the most activities)
+#     }
+# ]
+
+
```
Comment (larwaa) [inline src/mongodb/tasks.py]:
```suggestion
```

File: src/mongodb/test.py
Comment (larwaa) [inline src/mongodb/test.py:1]: Keep or remove

**Review by larwaa — CHANGES_REQUESTED** [review]
Looks great

File: src/mongodb/tasks.py (hunk 3 above)
Comment (sigurdnf) [reply src/mongodb/tasks.py:154]: Will fix later aligator

**Review by larwaa — APPROVED** [review]
LGTM 🌟

**Pull request merged**